builtin/pure_osh.py

```diff
@@ -382,6 +382,16 @@ def Run(self, cmd_val):
         arg = arg_types.hash(attrs.attrs)
 
         rest = arg_r.Rest()
+        if '-p' in attrs.attrs:
```

Comment:
To add a flag, see

```
HASH_SPEC = FlagSpec('hash')
HASH_SPEC.ShortFlag('-r')
```

in `frontend/flag_def.py`  .  And then regenerated code with `build/py.sh all`

https://github.com/oils-for-unix/oils/wiki/Oils-Dev-Cheat-Sheet

And then you can use `arg.p`

According to `help hash`, it is a flag that takes an argument, not 2 arguments.    `-p pathname`

So that's `ShortFlag('-p', args.String)`